src/compile/scale/component.ts

```diff
@@ -9,15 +9,13 @@ import {Explicit, Split} from '../split';
  * All VgDomain property except domain.
  * (We exclude domain as we have a special "domains" array that allow us merge them all at once in assemble.)
  */
-// TODO: also exclude domainRaw and property implement the right scaleComponent for selection domain
+// TODO: also exclude domainRaw and properly implement the right scaleComponent for selection domain
 
-export type ScaleComponentProps = Omit<VgScale, 'domain'>;
+export type ScaleComponentProps = Omit<VgScale, 'domain'> & {domains: VgNonUnionDomain[]};
```

Comment:
minor: technically, it's a component for scale domain. Given it's a domain of a single scale, then `domain` (singular) is more appropriate?

Reply:
Before parse, this is not the domain yet but an array of domains that gets merged unioned into one vega domain. The types are also incompatible so it's better to have a different name, imho.